openeo/extra/job_management/__init__.py

```diff
@@ -364,19 +371,30 @@ def run_loop():
             # TODO: support user-provided `stats`
             stats = collections.defaultdict(int)
 
-            while (
-                sum(job_db.count_by_status(statuses=["not_started", "created", "queued", "running"]).values()) > 0
-                and not self._stop_thread
-            ):
-                self._job_update_loop(job_db=job_db, start_job=start_job)
+            # TODO: multiple workers instead of a single one? Work with thread pool?
+            worker_thread = _JobManagerWorkerThread(work_queue=self._work_queue, result_queue=self._result_queue)
+            worker_thread.start()
+
+            while sum(job_db.count_by_status(statuses=["not_started", "created", "queued", "queued_for_start", "running"]).values()) > 0:
+                self._job_update_loop(job_db=job_db, start_job=start_job, stats=stats)
                 stats["run_jobs loop"] += 1
 
+                # Show current stats and sleep
                 _log.info(f"Job status histogram: {job_db.count_by_status()}. Run stats: {dict(stats)}")
-                # Do sequence of micro-sleeps to allow for quick thread exit
-                for _ in range(int(max(1, self.poll_sleep))):
-                    time.sleep(1)
-                    if self._stop_thread:
-                        break
+                time.sleep(1)
+                stats["sleep"] += 1
+
+            worker_thread.stop_event.set()
+            worker_thread.join()
+
+            _log.info(f"Job status histogram: {job_db.count_by_status()}. Run stats: {dict(stats)}")
+            # Do sequence of micro-sleeps to allow for quick thread exit
+            for _ in range(int(max(1, self.poll_sleep))):
+                time.sleep(1)
+                if self._stop_thread:
+                    break
```

Comment:
I'm still not sure I understand these changes. The batch of microsleeps still has to be inside the while loop I think (instead of that `sleep(1)` I think.